addons/mail/wizard/mail_followers_edit.py

```diff
@@ -0,0 +1,82 @@
+from odoo import fields, models
+from odoo.exceptions import UserError
+from odoo.addons.mail.tools.parser import parse_res_ids
+
+
+class MailFollowersEdit(models.TransientModel):
+    """Wizard to edit partners (or channels) to add/remove them to/from followers list."""
+
+    _name = 'mail.followers.edit'
+    _description = "Followers edit wizard"
+
+    res_model = fields.Char(
+        "Related Document Model", required=True, help="Model of the followed resource"
+    )
+    res_ids = fields.Char("Related Document IDs", help="Ids of the followed resources")
+    operation = fields.Selection(
+        [
+            ("add", "Add"),
+            ("remove", "Remove"),
+        ],
+        string="Operation",
+        required=True,
+        default="add",
+    )
+    partner_ids = fields.Many2many("res.partner", required=True, string="Followers")
+    message = fields.Html("Message")
+    notify = fields.Boolean("Notify Recipients", default=False)
+
+    def edit_followers(self):
+        for wizard in self:
+            res_ids = parse_res_ids(wizard.res_ids, self.env)
+            documents = self.env[wizard.res_model].browse(res_ids)
+            if wizard.operation == "remove":
+                for document in documents:
+                    document.message_unsubscribe(partner_ids=wizard.partner_ids.ids)
+            else:
+                if not self.env.user.email:
+                    raise UserError(
+                        self.env._(
+                            "Unable to post message, please configure the sender's email address."
+                        )
+                    )
+                for document in documents:
+                    # filter partner_ids to get the new followers, to avoid sending email to already following partners
+                    new_partners = wizard.partner_ids - document.sudo().message_partner_ids
+                    if not new_partners:
+                        continue
+                    document.message_subscribe(partner_ids=new_partners.ids)
+                    if wizard.notify:
```

Comment:
If somehow someone chooses 100 sale orders, add someone, then click "notify", do we really expect that person to receive 100 notifications ? Shouldn't we tailor a single one with all documents listed in it ?